File: exercice2_2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <errno.h>

int main()
{
	pid_t fils_1, fils_2 ;
	fils_1 = fork() ;
	if (fils_1 == -1)
	{
		perror("Problème dans fork lors de création de fils 1\n") ;
		exit (errno) ;
	}
	else if (fils_1 == 0)
	{
		printf("je suis le fils 1\n") ;
	}
	else
	{
		printf("je suis le père\n") ;
		wait(NULL) ;
		printf("fils 1 terminé\n");
		fils_2 = fork() ;
		if (fils_2 == -1)
		{
			perror("Problème dans fork lors de création de fils 2\n") ;
			exit (errno) ;
		}
		else if (fils_2 == 0)
		{
			printf("je suis le fils 2\n") ;
		}
		else
		{
			wait(NULL) ;
			printf("fils 2 terminé\n");

		}
	}
	return (0) ;
}

File: Python/exercice2_2.py
import os
import sys
import time

def main():
    
    fils_1 = os.fork()
    
    if fils_1 == -1:
        print("Problème dans fork lors de création de fils 1")
        sys.exit(1)
    
    elif fils_1 == 0:
        print("je suis le fils 1")
    
    else:
        print("je suis le père")
        
        os.wait()
        print("fils 1 terminé")
        
        fils_2 = os.fork()
        
        if fils_2 == -1:
            print("Problème dans fork lors de création de fils 2")
            sys.exit(1)
        
        elif fils_2 == 0:
            print("je suis le fils 2")
        
        else:
            os.wait()
            print("fils 2 terminé")

if __name__ == "__main__":
    main()